src/transaction/boot_sr.c

```diff
@@ -2659,7 +2659,7 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 #endif
 
 #if defined (SERVER_MODE)
-  if (get_server_type () == SERVER_TYPE_TRANSACTION)
+  if (is_active_transaction_server ())
```

Comment:
Compared to the develop branch, PTS also seems to have to call css_set_ha_num_of_hosts() and css_change_ha_server_state().